### PR TITLE
added CLAUDE.md

### PR DESCRIPTION
**Pull request by 23silicon**

### CODE REVIEW

File: CLAUDE.md [hunk 1]
```diff
@@ -0,0 +1,1078 @@
...
+**Format**: `"param_name": (ctype_spec, direction)`
+
+**Common ctypes**:
+- `ctypes.POINTER(ctypes.c_float)` - Float pointer (device memory)
+- `ctypes.POINTER(ctypes.c_int)` - Int pointer (device memory)
+- `ctypes.POINTER(ctypes.c_double)` - Double pointer (device memory)
+- `ctypes.c_float` - Single float value
+- `ctypes.c_int` - Integer value
+- `ctypes.c_size_t` - Size value (for array lengths)
+- `torch.Tensor` - PyTorch tensor (for frameworks that support it)
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: ino need to include torch.Tensor

File: CLAUDE.md [hunk 2]
```diff
@@ -0,0 +1,1078 @@
...
+- `torch.nn.Module` - Neural network module
+
+**Directions**:
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: theres also inout

File: CLAUDE.md [hunk 3]
```diff
@@ -0,0 +1,1078 @@
...
+        "B": B,
+        "C": C,
+        "N": N,
+    }
+```
+
+**Guidelines**:
+- Use small, simple values for clarity
+- Make example output manually verifiable
+- Should match or be similar to an example in `challenge.html`
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: should exactly match

File: CLAUDE.md [hunk 4]
```diff
@@ -0,0 +1,1078 @@
...
+- **Boundary values**: 0, 1, -1, INT_MAX/2, FLT_MAX/1000
+
+**Size Progression**:
+- **Named tests**: 1-1024 elements (comprehensive coverage)
+- **Random tests**: 32, 1000, 10000 elements (typical scales)
+- Total functional tests: ~12-15 test cases
+
+### Performance Test (`generate_performance_test`)
+
+Generates a single large test case to benchmark performance.
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: mention that 5x the size of the test case should fit comboftably in the VRAM of a TESLA T4 (16 GB)

File: CLAUDE.md [hunk 5]
```diff
@@ -0,0 +1,1078 @@
...
+### Reference Implementation (`reference_impl`)
+
+The reference implementation must:
+1. Accept the same parameters as the user's solution
+2. Perform the correct computation
+3. Include assertions for parameter validation (shape, dtype, device)
+4. Store results in output parameters (pass-by-reference)
+
+```python
+def reference_impl(self, A: torch.Tensor, B: torch.Tensor, C: torch.Tensor, N: int):
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: use the real vector add reference_impl

File: CLAUDE.md [hunk 6]
```diff
@@ -0,0 +1,1078 @@
...
+```
+
+**Guidelines**:
+- Specify size constraints (min/max)
+- Data type constraints
+- Value ranges (if applicable)
+- Special constraints (e.g., "matrices must be square")
+- Use HTML entities for math symbols: `&le;` (≤), `&ge;` (≥), `&ne;` (≠)
+
+#### 5. Optional: Time/Space Complexity (for medium/hard)
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: No. Please review your own PRs before you ask me to review them

File: CLAUDE.md [hunk 7]
```diff
@@ -0,0 +1,1078 @@
...
+
+```bash
+# Run tests
+python -m pytest tests/ -v
+
+# Check formatting
+pre-commit run --all-files
+```
+
+### Using generate_starter_code.py
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: don't mention this script.

File: CLAUDE.md [hunk 8]
```diff
@@ -0,0 +1,1078 @@
...
+```bash
+python scripts/generate_starter_code.py challenges/easy/5_my_challenge
+```
+
+Or create them manually, following the framework guidelines.
+
+#### Step 5: Validate & Test
+
+```bash
+# Run tests
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: nope

File: CLAUDE.md [hunk 9]
```diff
@@ -0,0 +1,1078 @@
...
+# - starter/starter.cute.py
+# - starter/starter.jax.py
+```
+
+**Generated starters**:
+- Easy: Include grid/block setup hints
+- Medium: Minimal starter structure
+- Hard: Just function signatures
+
+### Testing Challenges Locally
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: omit this section

File: CLAUDE.md [hunk 10]
```diff
@@ -0,0 +1,1078 @@
...
+
+All code must pass automated linting before merging. This is enforced by CI.
+
+### Python Code (`.py` files)
+
+**Tools**: black (formatting), isort (imports), flake8 (style)
+
+```bash
+# Install tools
+pip install black==24.1.1 isort==5.13.2 flake8==7.0.0
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: no, just link it to the contributing.md, dont have any information in this doc about formatting

File: CLAUDE.md (hunk 6 above)
Comment (nkotikal) [reply CLAUDE.md]: What should I delete here? The special constraints thing?

File: CLAUDE.md (hunk 7 above)
Comment (nkotikal) [reply CLAUDE.md]: generate_starter_code?

File: CLAUDE.md [hunk 11]
```diff
@@ -0,0 +1,524 @@
...
+7. [Creating & Testing Challenges](#creating--testing-challenges)
+8. [Formatting & Linting](#formatting--linting)
+9. [Directory Structure Checklist](#directory-structure-checklist)
+
+## Challenge Structure
+
+Each challenge must be placed in a specific directory hierarchy:
+
+```
+challenges/
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: Isn't this found in Contributing.md

File: CLAUDE.md [hunk 12]
```diff
@@ -0,0 +1,524 @@
...
+
+3. **Write challenge.html**
+   - Clear problem description
+   - 1-3 examples
+   - Precise constraints
+
+4. **Generate Starter Code**
+   ```bash
+   python scripts/generate_starter_code.py challenges/easy/<number>_challenge_name
+   ```
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: Don't use this script

File: CLAUDE.md (hunk 6 above)
Comment (kunal-mansukhani) [reply CLAUDE.md]: `#### 5. Optional: Time/Space Complexity (for medium/hard)`

File: CLAUDE.md (hunk 7 above)
Comment (kunal-mansukhani) [reply CLAUDE.md]: yes

File: CLAUDE.md [hunk 13]
```diff
@@ -0,0 +1,506 @@
...
+    return {
+        "A": (ctypes.POINTER(ctypes.c_float), "in"),
+        "B": (ctypes.POINTER(ctypes.c_float), "in"),
+        "C": (ctypes.POINTER(ctypes.c_float), "out"),
+        "N": (ctypes.c_size_t, "in"),
+    }
+```
+
+**Common ctypes**: `ctypes.POINTER(ctypes.c_float)`, `ctypes.c_int`, `ctypes.c_size_t`
+**Directions**: `"in"`, `"out"`, `"inout"`
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: explain what `inout` is

File: CLAUDE.md [hunk 14]
```diff
@@ -0,0 +1,506 @@
+# LeetGPU Challenge Creation Guide
+
+This guide provides instructions for adding new challenges to the LeetGPU challenge set, covering structure, metadata, test cases, and best practices.
+
+## Table of Contents
```
Comment (kunal-mansukhani) [inline CLAUDE.md]: dont think we need a table of contents yet. an agent will be able to injest all of this in 1 shot

Comment (nkotikal) [reply CLAUDE.md]: fixed

Comment (nkotikal) [reply CLAUDE.md]: clear to merge or any more changes?
